bottlerocket-settings-models/modeled-types/src/shared.rs

```diff
@@ -1327,3 +1327,69 @@ mod test_kernel_cpu_set_value {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// SizeString represents a string that contains a valid size specification with optional units.
+/// Valid formats include: "8b", "16kb", "64mb", "1gb", "1.5gb", "100", etc.
+/// The regex pattern is case-insensitive and allows optional whitespace.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct SizeString {
+    inner: String,
+}
```

Comment:
May be nice to add a method on this to get the configured size as a [uom::si::information](https://crates.io/crates/uom) type. Could help to "re-use" this information elsewhere and make the implementation more generic.

Not a blocker though, for now we know exactly where this will be used and what the string form must be.